v2/pkg/contextargs/contextargs.go

```diff
@@ -0,0 +1,19 @@
+package contextargs
+
+import (
+	"net/http/cookiejar"
+	"sync"
+)
+
+type Context struct {
+	Input     string
+	CookieJar *cookiejar.Jar
+
+	// Access to Args must use lock strategies to prevent data races
+	*sync.RWMutex
```

Comment:
I don't think it will be used that often, that's why I believe that RWMutex gives more flexibility as it will not lock completely, but still allow concurrent read operations